modules/dnn/CMakeLists.txt

```diff
@@ -249,6 +249,10 @@ ocv_create_module(${libs} ${dnn_runtime_libs})
 ocv_add_samples()
 ocv_add_accuracy_tests(${dnn_runtime_libs})
 
+if(NOT BUILD_PROTOBUF)
+  ocv_target_compile_definitions(opencv_test_dnn PRIVATE "OPENCV_DNN_EXTERNAL_PROTOBUF=1")
```

Comment:
Does this need "if(TARGET opencv_test_dnn)" as well? I'm seeing:

```
CMake Error at cmake/OpenCVUtils.cmake:1517 (message):
  ocv_target_compile_definitions: invalid target: 'opencv_test_dnn'
Call Stack (most recent call first):
  modules/dnn/CMakeLists.txt:258 (ocv_target_compile_definitions)
```

Reply:
Hello, thank you for your comment, I agree with you.
When `ENABLE_TESTS=OFF`, same problem is occured.
I push additional patch to fix it.